lib/log.bash

```diff
@@ -1,60 +1,90 @@
-#!/usr/bin/env bash
+# shellcheck shell=bash
+#
+# A collection of logging functions.
 
-export BASH_IT_LOG_LEVEL_ERROR=1
-export BASH_IT_LOG_LEVEL_WARNING=2
-export BASH_IT_LOG_LEVEL_ALL=3
+# Declare log severity levels, matching syslog numbering
+: "${BASH_IT_LOG_LEVEL_FATAL:=1}"
```

Comment:
do we really need those? lets at least define useful log functions for these

Reply:
We don't need these, but I just threw them in since I was harmonizing the numbers with `syslog`. I think five log levels make sense for _Bash It_: error, warning, notice, debug, and trace. 

- Error: something is broken and doesn't work. 
- Warning: something is unexpected but may still be fine. 
- Notice: something happened that should be noted. 
- Info: unnecessary details, useful for debugging. 
- Trace: annoying detail, useful only for serious debugging.

Reply:
(A future PR from me touches `lib/log` again to provide continuous logging to a file, with optional echo to screen, and optional integration with _Bash_'s other logging features and even `syslog` if configured.)